File: api/src/main/java/org/egglog/api/hospital/model/dto/response/UserHospitalResponse.java
package org.egglog.api.hospital.model.dto.response;

import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.*;

/**
 * packageName    : org.egglog.api.hospital.model.dto.response
 * fileName       : HospitalResponse
 * author         : 김형민
 * date           : 2024-04-22
 * description    : 병원 정보를 리턴하는 객체입니다.
 * ===========================================================
 * DATE              AUTHOR             NOTE
 * -----------------------------------------------------------
 * 2024-04-22        김형민       최초 생성
 */
@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
public class UserHospitalResponse {

    private Long id;
    private String sidoCode;
    private String sido;
    private String gunguCode;
    private String gungu;
    private String dong;
    private String zipCode;
    private String address;
    private String hospitalName;
    private String lat;
    private String lng;

}

File: client/data/src/main/java/com/org/egglog/data/auth/model/UserHospitalResponse.kt
package com.org.egglog.data.auth.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import com.org.egglog.domain.auth.model.UserHospital
import kotlinx.serialization.Serializable

@Entity
@Serializable
data class UserHospitalResponse(
    @PrimaryKey val hospitalId: Long,
    val sidoCode: String?,
    val sido: String?,
    val gunguCode: String?,
    val gungu: String?,
    val dong: String?,
    val zipCode: String?,
    val address: String,
    val hospitalName: String,
    val lat: String?,
    val lng: String?
)

fun UserHospitalResponse.toDomainModel(): UserHospital {
    return UserHospital(
        hospitalId, sidoCode, sido, gunguCode, gungu, dong, zipCode, address, hospitalName, lat, lng
    )
}